### PR TITLE
[DDW-81] Improve redirecting logic of network store

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR greatly simplifies the network status store and improves the booting UX:

* Removes any redirecting / URL / screen handling logic from `NetworkStatusStore`
* Introduces new `Root` route + component which handles the conditional logic of what to display based on various states
* Show the loading spinner while waiting for the time difference request to finish (instead of showing 100% syncing message for 5 seconds)
* The original URL is now preserved (because we are not doing any redirects), this means that you can now reload the app on sub-screens like "wallet / send" and it will return to the same 😉 

All tests pass as before, also tested everything with `API=etc` and there are not lint or flow issues.

### CODE REVIEW

**Comment by DominikGuzei** [comment]
@nikolaglumac this is now also ready to be reviewed!

File: source/renderer/app/containers/etc/LoadingPage.js [hunk 1]
```diff
@@ -32,7 +31,7 @@ export default class LoadingPage extends Component<InjectedProps> {
           allowedTimeDifference={ALLOWED_TIME_DIFFERENCE}
           isConnecting={isConnecting}
           syncPercentage={syncPercentage}
-          isLoadingDataForNextScreen={isLoadingWallets}
+          isLoadingDataForNextScreen={!isSyncing || isSynced}
```
Comment (nikolaglumac) [inline source/renderer/app/containers/etc/LoadingPage.js]: @DominikGuzei I believe we should take this opportunity to refactor `isSyncing` and `isSynced` logic in the `NetworkStatusStore`: https://github.com/input-output-hk/daedalus/blob/development-webpack-fork/source/renderer/app/stores/NetworkStatusStore.js#L140-L151

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 2]
```diff
@@ -47,43 +46,53 @@ export default class NetworkStatusStore extends Store {
...
-  _timeDifferencePollInterval: ?number = null;
-
   @action initialize() {
     super.initialize();
-    if (cachedDifficulties !== null) Object.assign(this, cachedDifficulties);
+    if (cachedState !== null) Object.assign(this, cachedState);
   }
 
-  setup() {
+  async setup() {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DominikGuzei can you explain what is the reason for `async` here?

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 3]
```diff
@@ -47,43 +46,53 @@ export default class NetworkStatusStore extends Store {
...
+      this._updateLocalTimeDifferenceWhenConnected,
     ]);
-    this._pollSyncProgress();
+
+    // Setup polling intervals
+    this._syncProgressPollInterval = setInterval(
+      this._updateSyncProgress, SYNC_PROGRESS_INTERVAL
+    );
+    if (environment.isAdaApi()) {
+      this._updateLocalTimeDifferencePollInterval = setInterval(
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DominikGuzei I fear this update is in collision with the following improvement: https://github.com/input-output-hk/daedalus/pull/719

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 4]
```diff
@@ -218,85 +224,15 @@ export default class NetworkStatusStore extends Store {
...
-      this._updateLocalTimeDifference, TIME_DIFF_POLL_INTERVAL
-    );
-    this._updateLocalTimeDifference();
-  }
-
-  _stopPollingLocalTimeDifference() {
-    Logger.debug('Stopped polling local time difference');
-    if (this._timeDifferencePollInterval) clearInterval(this._timeDifferencePollInterval);
+  _updateLocalTimeDifferenceWhenConnected = async () => {
+    if (this.isConnected) await this._updateLocalTimeDifference();
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @DominikGuzei this call should be made only in case of `environment.isAdaApi()`

**Comment by nikolaglumac** [comment]
@DominikGuzei can you please merge in latest develop in your branch? I will review this PR after you do that. Thanks!

**Comment by nikolaglumac** [comment]
@DominikGuzei please merge in latest develop so that this one can finally be reviewed :) Thanks!

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 3 above)
Comment (DominikGuzei) [reply source/renderer/app/stores/NetworkStatusStore.js]: No it doesn't the code is even better now, you have to read through it carefully 😉

**Comment by DominikGuzei** [comment]
@nikolaglumac i fixed the conflicts with latest `develop` branch and all tests pass! Please review carefully as these are nice improvements.

**Comment by DominikGuzei** [comment]
@nikolaglumac please test again, i hopefully fixed all issues with the loading screens now!

**Comment by nikolaglumac** [comment]
Let's wait with merging this PR until the 1.3 release cut off date.

**Comment by DominikGuzei** [comment]
@nikolaglumac this is ready for review (again)

**Comment by nikolaglumac** [comment]
@DominikGuzei reviewing it now! (again)

File: source/renderer/app/stores/ada/AdaRedemptionStore.js [hunk 5]
```diff
@@ -62,6 +62,7 @@ export default class AdaRedemptionStore extends Store {
     ipcRenderer.on(PARSE_REDEMPTION_CODE.ERROR, this._onParseError);
     this.registerReactions([
       this._resetRedemptionFormValuesOnAdaRedemptionPageLoad,
+      this._redirectToAddWalletBeforeRedemption,
```
Comment (nikolaglumac) [inline source/renderer/app/stores/ada/AdaRedemptionStore.js]: @DominikGuzei I don't think we need this redirect anymore as handle this case by showing special content on the Ada redemption screen.

File: source/renderer/app/stores/ada/AdaRedemptionStore.js [hunk 6]
```diff
@@ -299,4 +300,12 @@ export default class AdaRedemptionStore extends Store {
     this.decryptionKey = null;
   };
 
+  _redirectToAddWalletBeforeRedemption = () => {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/ada/AdaRedemptionStore.js]: @DominikGuzei this is not needed anymore!

**Comment by nikolaglumac** [comment]
@DominikGuzei awesome work! This PR is ready to be merged as soon as CI is done ;)

**Pull request merged**